sympy/geometry/line.py

```diff
@@ -1070,6 +1070,40 @@ def contains(self, other):
             return Point.is_collinear(self.p1, self.p2, other.p1, other.p2)
         return False
 
+    def coordinate(self, other):
+        """
+        Finds the parameter ``t`` for which
+        ``other == t * self.p1 + (1 - t) self.p2`` is True.
+
+        Raises
+        ======
+
+        ValueError
+            When ``other`` is not contained in the line.
```

Comment:
I would like to see the NotImplementedError converted to a ValueError. So ValueError would be raised if other is not a Point.